fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt

```diff
@@ -4,4 +4,38 @@
 
 package mozilla.components.service.fretboard
 
-class Fretboard
+/**
+ * Entry point of the library
+ */
+class Fretboard {
+    companion object {
+        private lateinit var client: HttpClient
+        private lateinit var source: ExperimentSource
+        private lateinit var storage: ExperimentStorage
+
+        /**
+         * Configures Fretboard to use the given http client,
+         * experiment source and local storage mechanism
+         *
+         * @param client http client to use
+         * @param source experiment remote source
+         * @param storage experiment local storage mechanism
+         */
+        @JvmStatic
+        fun configure(client: HttpClient, source: ExperimentSource, storage: ExperimentStorage) {
+            Fretboard.client = client
+            Fretboard.source = source
+            Fretboard.storage = storage
+        }
+
+        /**
+         * Requests new experiments from the server and
+         * saves them to local storage
+         */
+        @JvmStatic
+        fun loadExperiments() {
+            val experiments = source.getExperiments(client, storage.retrieve())
+            storage.save(experiments)
+        }
+    }
+}
```

Comment:
Let's make `Fretboard` something you can actually instantiate (instead of static methods), like:

```
val fretboard = Fretboard(client, storage)
```

Leave it to the consumer of the library to decide how the instance should be kept around.

Reply:
Ok, makes more sense. Will change it

Reply:
Done :)